tests/NHSD.BuyingCatalogue.Solutions.Persistence.DatabaseTests/SupplierRepositoryTests.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Moq;
+using NHSD.BuyingCatalogue.Solutions.Contracts.Persistence;
+using NHSD.BuyingCatalogue.Testing.Data;
+using NHSD.BuyingCatalogue.Testing.Data.Entities;
+using NHSD.BuyingCatalogue.Testing.Data.EntityBuilders;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Solutions.Persistence.DatabaseTests
+{
+    [TestFixture]
+    public sealed class SupplierRepositoryTests
+    {
+        private ISupplierRepository _supplierRepository;
+        private readonly Guid _org1Id = Guid.NewGuid();
+        private readonly string _supplierId = "Sup 1";
+        private readonly string _solutionId = "Sln1";
+
+        private readonly string _description = "some description";
+        private readonly string _link = "www.someLink.com";
+
+        private readonly string _newDescription = "new description";
+        private readonly string _newLink = "www.newLink.com";
+
+        [SetUp]
+        public async Task SetUp()
+        {
+            await Database.ClearAsync().ConfigureAwait(false);
+
+            await OrganisationEntityBuilder.Create()
+                .WithId(_org1Id)
+                .Build()
+                .InsertAsync()
+                .ConfigureAwait(false);
+
+            TestContext testContext = new TestContext();
+            _supplierRepository = testContext.SupplierRepository;
+        }
+
+        [Test]
+        public async Task ShouldGetSupplier()
+        {
+            var expected = await InsertSupplier().ConfigureAwait(false);
+
+            var result = await _supplierRepository.GetSupplierBySolutionIdAsync(_solutionId, new CancellationToken())
+                .ConfigureAwait(false);
+
+            result.SolutionId.Should().Be(_solutionId);
+            result.Description.Should().Be(_description);
+            result.Link.Should().Be(_link);
+        }
+
+        [Test]
+        public async Task ThrowNotFoundOnSupplierIfNoSupplierOrSolution()
+        {
```

Comment:
Is the name of the method misleading here? Where are we asserting the throwing of the not found exception?